### ISSUE TITLE
Mimir mixin doesn't support high scrape intervals

### ISSUE THREAD

**Issue by replay** [grafana/mimir#5509]
#### Describe the bug
If the Mimir metrics are scraped with a high scrape interval, for example `1m`, many queries in the mixin break because their range vector selector is too low for the sample interval.

#### To Reproduce

* Install the mimir mixin
* Scrape Mimir metrics with a scrape interval of `1m`
* Various dashboard panels don't show data because there are not enough samples within each `1m` range vector

#### Expected behavior

The dashboard panels are still supposed to show data. 
Optimally the mixin would have a configuration parameter that allows the user to define their scrape interval, then the range vectors in the queries would be raised to a value that's high enough to ensure that the queries succeed.
Alternatively, the simpler solution is to just change the hard coded range vectors that are currently `1m` to `5m`. The resulting panels would still be useful and it would solve the problem, as we can assume that nobody will use a scrape interval of `>5m`

**Comment by pstibrany** [comment]
> as we can assume that nobody will use a scrape interval of `>5m`

We thought the same about `1m` :(

**Comment by doanbutar** [comment]
Hello @replay May I follow up on this issue, please? Can we try the simpler solution to change the hard-coded range to `5m`?

**Comment by replay** [comment]
Yes, if the user would increase their scrape rate then this problem would be solved for them. 

However, I still think that the "proper" solution to this problem is to make the mixin handle any scrape rate that the user wants to configure because not doing so is just unnecessarily complicating the setup of Mimir + mixin, without any real benefit. I don't think there's a good argument to be made why the scrape interval should always be below `1m`, that's just up to the user to figure out what makes sense for them.

**Comment by 09jvilla** [comment]
> I don't think there's a good argument to be made why the scrape interval should always be below 1m, that's just up to the user to figure out what makes sense for them.

Understood. But I think the core maintainers of Mimir would argue that a scrape interval of 10m would not be very useful (or at least, not in accordance with recommended best practices), so I don't know that the mixin needs to necessarily support arbitrary scrape intervals.

**Comment by doanbutar** [comment]
Hi @replay and @09jvilla, thanks for looking into this. 
Do you mean `1m` instead of `10m` in your previous comment @09jvilla ? 

For context: The customer just moved their scrape_interval for mimir metrics to **60s**, in line with the rest of their environment; many of the mimir-mixin (https://github.com/grafana/mimir/tree/main/operations/mimir-mixin) recording rules and dashboards broke.

They would've expected configuration to allow for adjusting supported ranges https://github.com/grafana/mimir/blob/main/operations/mimir-mixin/config.libsonnet#L639 handles some but not all cases.

**Comment by 09jvilla** [comment]
> Do you mean 1m instead of 10m in your previous comment @09jvilla ?
No, I meant 10m to make the point that while it makes sense to support 1m scrape intervals, that doesn't mean we should support or encourage arbitrary scrape intervals (like once every 10m or once every 30m, which are both too infrequent to be of any use). 

> They would've expected configuration to allow for adjusting supported ranges
Understand their expectation. 
You can tell them that we use a more frequent scrape interval ourselves, which is why the mixin doesn't support their less frequent scrape interval today. We find the more frequent scrape interval helpful to debug problems with Mimir faster (since you're getting data more frequently and can see changes more frequently). 

That being said, we recognize others may only want to do one scrape per minute; they're ok with lower resolution and/or waiting longer to see a new datapoint in exchange for saving on cost (since they have fewer datapoints). 

We can close this issue and say we've created a feature request to make the mixin more flexible in supporting other scrape rates. We've added it to our backlog. No plans to pick it up for the next ~2 weeks, but they can check in with us via their CSM after that if desired for an update.

**Comment by chap-dr** [comment]
I know this issue is closed, And I am not looking to reopen it, just wanted to leave a comment saying that you hit this issue if you use the k8s-monitoring default values for scrapeinterval which is set to 60s. https://github.com/grafana/k8s-monitoring-helm/blob/main/charts/k8s-monitoring/values.yaml#L18
